cmake/macos/compilerconfig.cmake

```diff
@@ -14,6 +14,10 @@ include(compiler_common)
 
 add_compile_options("$<$<NOT:$<COMPILE_LANGUAGE:Swift>>:-fopenmp-simd>")
 
+if(CMAKE_CXX_STANDARD GREATER_EQUAL 20)
+  add_compile_options($<$<COMPILE_LANGUAGE:CXX>:-fno-char8_t>)
+endif()
```

Comment:
I'd actually like us to opt-in to the new type and make it explicit that strings are explicitly required to be UTF-8 encoded and not "whatever".

But I'd probably agree that the scope of that change deserves its own PR. 😅

Reply:
I think a big issue with that is that many STL-libraries don't even support `std::u8string`, not even mentioning any third-party libraries we're using (e.g. Qt). Agree it would be nice if we could though.
But yeah out of scope here (and would break C++17 support? So probably work for after the switch has been changed to use C++20).